spec.md

```diff
@@ -365,15 +368,68 @@ allowing a long string to be split across multiple lines of the source file.
 def"			# "abcdef"
 ```
 
-An *octal escape* encodes a single byte using its octal value.
+An *octal escape* encodes a single string element using its octal value.
 It consists of a backslash followed by one, two, or three octal digits [0-7].
-It is error if the value is greater than decimal 255.
+Simiarly, a *hexadecimal escape* encodes a single string element using its hexadecimal value.
+It consists of `\x` followed by two hexadecimal digits [0-9a-fA-F].
+It is an error if the value of an octal or hexadecimal escape is greater than decimal 127.
 
 ```python
-'\0'			# "\x00"  a string containing a single NUL byte
+'\0'			# "\x00"  a string containing a single NUL element
 '\12'			# "\n"    octal 12 = decimal 10
 '\101-\132'		# "A-Z"
 '\119'			# "\t9"   = "\11" + "9"
+
+'\x00'			# "\x00"  a string containing a single NUL element
+'\0A'			# "\n"    hexadecimal A = decimal 10
+"\x41-\x5A"             # "A-Z"
+```
+
+The bit width of each string element is not defined by this spec,
+and for reasons of efficiency and interoperability with the host language,
+it varies across implementations.
+For example, in the Go and Rust implementations,
+each string element is an 8-bit value (a byte) and Unicode text is encoded as UTF-8,
+whereas in the Java implementation,
+string elements are 16-bit values (Java `char`s) and Unicode text is encoded as UTF-16.
+This spec generalizes both approaches by referring to the string encoding as "UTF-K",
+where the value of K is implementation-defined.
+
+A *Unicode escape* denotes the UTF-K encoding of a single Unicode code point.
+The `\uXXXX` form, with exactly four hexadecimal digits,
+denotes a 16-bit code point, and the `\UXXXXXXXX`,
+with exactly eight digits, denotes a 32-bit code point.
+It is an error if the value lies in the surrogate range (U+D800 to U+DFFF).
+
+```python
+'\u0041'		# "A", an ASCII letter (U+0041)
+'\u0414' 		# "Д", a Cyrillic capital letter (U+0414)
+'\u754c                 # "界", a Chinese character (U+754C)
+'\U0001F600'            # "😀", an Emoji (U+1F600)
+```
+
+The length of the encoding of a single Unicode code point may vary
+based on the implementation's string encoding ("K").
+
+```python
+len("A") 		# 1
+len("Д") 		# 2 (UTF-8) or 1 (UTF-16)
+len("界")               # 3 (UTF-8) or 1 (UTF-16)
+len("😀")               # 4 (UTF-8) or 2 (UTF-16)
+```
+
+An *element escape* denotes a single element of a text string.
```

Comment:
Comment out or delete this part as per yesterday's discussion.

Reply:
Done.